cobalt/browser/cobalt_content_browser_client.cc

```diff
@@ -27,14 +28,75 @@
 // TODO(b/390021478): Remove this include when CobaltBrowserMainParts stops
 // being a ShellBrowserMainParts.
 #include "content/shell/browser/shell_browser_main_parts.h"
+#include "services/network/public/mojom/network_context.mojom.h"
+#include "starboard/configuration_constants.h"
+#include "starboard/system.h"
 #include "third_party/blink/public/common/web_preferences/web_preferences.h"
+#include "third_party/icu/source/common/unicode/uloc.h"
 
 #if BUILDFLAG(IS_ANDROIDTV)
 #include "cobalt/browser/android/mojo/cobalt_interface_registrar_android.h"
 #endif
 
+#if BUILDFLAG(IS_LINUX)
+#include "components/os_crypt/sync/key_storage_config_linux.h"
+#include "components/os_crypt/sync/os_crypt.h"
+#endif
+
 namespace cobalt {
 
+namespace {
+
+#define FPL FILE_PATH_LITERAL
+const base::FilePath::CharType kCacheDirname[] = FPL("Cache");
+const base::FilePath::CharType kCookieFilename[] = FPL("Cookies");
+const base::FilePath::CharType kNetworkDataDirname[] = FPL("Network");
+const base::FilePath::CharType kNetworkPersistentStateFilename[] =
+    FPL("Network Persistent State");
+const base::FilePath::CharType kSCTAuditingPendingReportsFileName[] =
+    FPL("SCT Auditing Pending Reports");
+const base::FilePath::CharType kTransportSecurityPersisterFilename[] =
+    FPL("TransportSecurity");
+const base::FilePath::CharType kTrustTokenFilename[] = FPL("Trust Tokens");
+#undef FPL
+
+base::FilePath GetCacheDirectory() {
+  std::vector<char> path(kSbFileMaxPath, 0);
+  if (!SbSystemGetPath(kSbSystemPathCacheDirectory, path.data(),
+                       kSbFileMaxPath)) {
+    NOTREACHED() << "Failed to get cache directory.";
+    return base::FilePath();
+  }
+  return base::FilePath(path.data());
+}
+
+#define arraysize(f) (sizeof(f) / sizeof(*f))
```

Comment:
Let's try to avoid a macro, either by inlining it in its two uses or using a function like

```
template <size_t N>
constexpr size_t arraysize(const char (&arr)[N]) {
    return N;
}
```